pkg/controlplane/apiserver/apis.go

```diff
@@ -134,6 +195,23 @@ func (s *Server) InstallAPIs(restStorageProviders ...RESTStorageProvider) error
 			// everything else goes to /apis
 			nonLegacy = append(nonLegacy, &apiGroupInfo)
 		}
+
+		for version, storageMap := range apiGroupInfo.VersionedResourcesStorageMap {
+			for resource, storage := range storageMap {
+				if withReadiness, ok := storage.(rest.StorageWithReadiness); ok {
+					gvr := metav1.GroupVersionResource{
+						Group:    groupName,
+						Version:  version,
+						Resource: resource,
+					}
+					storageReadinessHook.RegisterStorage(gvr, withReadiness)
+				}
+			}
+		}
```

Comment:
Worth feature gating this?  I don't feel strongly, but noticed it while reading.

Reply:
registering post-start-hook is gated below - gating this here won't buy us anything more, right?

Reply:
It would take more code out of the code path, but otherwise no.